Add typescript to library

danielwinkler opened the pull request.

gitstud commented:
Let's get these conflicts resolved and merged 💃 :)

danielwinkler commented:
@gitstud this one is fixed  too!

jasongaare commented on src/WalkthroughElement.tsx:
```diff
@@ -0,0 +1,64 @@
+
+import {WalkthroughContext} from './ContextWrapper';
+
+type UseTooltipChildContextT =
```
this type should have a lower case `t` at the end, right?

danielwinkler replied:
the T should have indicated that it's a type, but i think without the T it's also fine as there is no name conflict

jasongaare commented on .prettierrc:
```diff
@@ -0,0 +1,10 @@
+  "singleQuote": true,
+  "trailingComma": "es5",
+  "arrowParens": "avoid",
+  "bracketSpacing": false
```
personal pref, but could you set "bracketSpacing" to `true` please?

danielwinkler replied:
also my preference, i changed it and reformatted the code :-)

gitstud commented:
@jasongaare This should help for publishing https://medium.com/cameron-nokes/the-30-second-guide-to-publishing-a-typescript-package-to-npm-89d93ff7bccd

danielwinkler commented:
@jasongaare I implemented your suggestions

i also added tsdx which should make the whole process quite easy, 
https://github.com/jaredpalmer/tsdx

I think it'll be sufficient to read the [Quick start](https://github.com/jaredpalmer/tsdx)

just build your TS project and it will create everything you need in the dist folder:
`yarn build
yarn run v1.21.1
$ tsdx build --format cjs,esm,umd --name react-native-walkthrough
✓ Creating entry file 2.8 secs
⠸ Building modules  54%                 1.8s, estimated 3.4s
No name was provided for external module 'react-native-walkthrough-tooltip' in output.globals – guessing 'Tooltip'
No name was provided for external module 'prop-types' in output.globals – guessing 'PropTypes'
No name was provided for external module 'events' in output.globals – guessing 'events'
Creating a browser bundle that depends on Node.js built-in module ('events'). You might need to include https://www.npmjs.com/package/rollup-plugin-node-builtins
No name was provided for external module 'react-native-walkthrough-tooltip' in output.globals – guessing 'Tooltip'
No name was provided for external module 'prop-types' in output.globals – guessing 'PropTypes'
No name was provided for external module 'events' in output.globals – guessing 'events'
✓ Building modules 2.6 secs
✨  Done in 6.54s.`

Then you can publish to npm (it would trigger the build above automatically if you forgot):
`yarn publish
yarn publish v1.21.1
[1/4] Bumping version...
info Current version: 0.1.0
question New version: 0.1.1
info New version: 0.1.1
[2/4] Logging in...`

danielwinkler commented:
@gitstud Thanks for your support !

jasongaare commented:
I apologize it's been hard for me to find time for this project lately...

Finally released as version `0.2.0` 🎉

danielwinkler commented:
@jasongaare 

no worries, I really appreciate your work and totally understand that it's a hard task to maintain a side project!

Cheers,
Daniel
